packages/core/src/comparator.test.js

```diff
@@ -148,7 +148,7 @@ describe("compare date fields", () => {
     secondDate.prop("onFocus")();
     form.update();
     expect(form.state().fields[0].errorMessages).toBe(
-      "A value must be provided"
+      "A value must be provided, Must be before second date"
```

Comment:
The additional error message now shows because although a value hasn't been provided it still fails the basic requirement test so the additional message is shown. This could potentially be improved - but if we're going to show one error message we may as well show both.